travis-ci/run-standard-tests.sh

```diff
@@ -41,33 +35,55 @@ else
     ulimit -Sn 8192
 fi
 
-# Test
-if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+if [[ $GXBUILD ]]; then
+    echo "*** Installing gx"
+    display_and_run go get github.com/whyrusleeping/gx
+    display_and_run go get github.com/whyrusleeping/gx-go
+    export GO111MODULE=off
+    display_and_run gx install --nofancy
+    display_and_run gx-go rw
+fi
+
+
+list_buildable() {
+    go list -f '{{if (len .GoFiles)}}{{.ImportPath}} {{if .Module}}{{.Module.Dir}}{{else}}{{.Dir}}{{end}}{{end}}' ./... | grep -v /vendor/
+}
+
+build_all() {
     # Make sure everything can compile since some package may not have tests
     # Note: that "go build ./..." will fail if some packages have only
     #   tests (will get "no buildable Go source files" error) so we
     #   have to do this the hard way.
-    go list -f '{{if (len .GoFiles)}}{{.ImportPath}} {{if .Module}}{{.Module.Dir}}{{else}}{{.Dir}}{{end}}{{end}}' ./... | grep -v /vendor/ | \
-        while read pkg dir; do
-            (
-              cd "$dir"
-              go build -o /dev/null "$pkg"
-            )
-        done
-    echo
+    list_buildable | while read -r pkg dir; do
+        echo '*** go build' "$pkg"
+        ( cd "$dir"; go build -o /dev/null "$pkg")
+    done
+}
+
+list_testable() {
+    go list -f '{{if or (len .TestGoFiles) (len .XTestGoFiles)}}{{.ImportPath}}{{end}}' ./... | grep -v /vendor/ || true
+}
+
+# Test
+if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+    # build all packages in the repo
+    build_all
+
+    list_testable > packages-with-tests
     # Run tests with coverage report in each packages that has tests
-    go list -f '{{if or (len .TestGoFiles) (len .XTestGoFiles)}}{{.ImportPath}}{{end}}' ./... | grep -v /vendor/ > packages-with-tests || true
     if [[ -s packages-with-tests ]]; then
-        cat packages-with-tests | while read pkg; do
+        while read -r pkg; do
             profile="coverage.$(echo "$pkg" | md5sum | head -c 16).txt"
-            display_and_run go test -v $GOTFLAGS -coverprofile="$profile" -covermode=atomic "$pkg"
-        done
+            display_and_run go test -v "${GOTFLAGS[@]}" -coverprofile="$profile" -covermode=atomic "$pkg"
+        done < packages-with-tests
+
         # Doesn't count as a failure.
         echo "*** Processing coverage report"
         bash <(curl -s https://codecov.io/bash) || true
```

Comment:
It doesn't count as a failure but it would be useful to know if the coverage wasn't pushed to codecov, so just a log message would be good.